git-repo.php

```diff
@@ -728,3 +768,497 @@ function vipgoci_gitrepo_submodule_get_url(
+		 */
+		if (
+			( $cur_file_path_cleaned === false ) &&
+			( null !== $cur_file )
+		) {
+			$cur_file = vipgoci_sanitize_path_prefix(
+				$cur_file,
+				array( 'a/', 'b/' )
+			);
+
+			$cur_file_path_cleaned = true;
+		}
+
+		if (
+			( $cur_file_minus_path_cleaned === false ) &&
+			( null !== $cur_file_minus )
+		) {
+			$cur_file_minus = vipgoci_sanitize_path_prefix(
+				$cur_file_minus,
+				array( 'a/' )
+			);
+
+			$cur_file_minus_path_cleaned = true;
+		}
+
+		if (
+			( $cur_file_plus_path_cleaned === false ) &&
+			( null !== $cur_file_plus )
+		) {
+			$cur_file_plus = vipgoci_sanitize_path_prefix(
+				$cur_file_plus,
+				array( 'b/' )
+			);
+
+			$cur_file_plus_path_cleaned = true;
+		}
+
+
+		/*
+		 * Logic to handle file names.
+		 */
+		if (
+			( ! empty( $cur_file_minus ) ) ||
+			( ! empty( $cur_file_plus ) )
+		) {
+			if ( $cur_file_minus === $cur_file_plus ) {
+				$cur_file = $cur_file_plus;
+				$cur_file_status = 'modified';
+			}
+
+			else {
+				/*
+				 * If the file names do not match,
+				 * and either is /dev/null, then
+				 * it is a new file or a removed file.
+				 */
+				if ( $cur_file_minus === '/dev/null' ) {
+					$cur_file = $cur_file_plus;
+					$cur_file_status = 'added';
+				}
+
+				else if ( $cur_file_plus === '/dev/null' ) {
+					$cur_file = $cur_file_minus;
+					$cur_file_status = 'removed';
+				}
+
+				/*
+				 * No match and no mention of /dev/null,
+				 * so the file must have been renamed.
+				 */
+				else {
+					$cur_file = $cur_file_plus;
+					$cur_file_status = 'renamed';
+				}
+			}
+		}
+
+		/*
+		 * If the file is not yet registered in the
+		 * 'diff' array, do so now. This will include
+		 * files that only have permission changes.
+		 */
+		if (
+			( null !== $cur_file ) &&
+			( null !== $cur_file_status ) &&
+			( ! isset( $diff_results['files'][ $cur_file ] ) )
+		) {
+			$diff_results['files'][ $cur_file ] = array(
+				'filename'	=> $cur_file,
+				'patch'		=> '',
+				'status'	=> $cur_file_status,
+				'additions'	=> 0,
+				'deletions'	=> 0,
+				'changes'	=> 0,
+			);
+		}
+
+		if ( null !== $cur_file_status ) {
+			/*
+			 * Update file-status each time we loop
+			 * as the calculated status might change.
+			 */
+			$diff_results['files'][ $cur_file ][ 'status' ] =
+				$cur_file_status;
+		}
+
+		if ( 'patch' !== $cur_mode ) {
+			/* Not in patch mode, so do not collect patch */
+			continue;
+		}
+
+		/*
+		 * Sanity check.
+		 */
+		if ( empty( $cur_file ) ) {
+			vipgoci_log(
+				'Problem when getting git diff, no file name found in patch',
+				array(
+					'local_git_repo'	=> $local_git_repo,
+					'commit_id_a'		=> $commit_id_a,
+					'commit_id_b'		=> $commit_id_b,
+					'cur_file'		=> $cur_file,
+					'cur_file_minus'	=> $cur_file_minus,
+					'cur_file_plus'		=> $cur_file_plus,
+
+				)
+			);
+
+			continue;
+		}
+
+		/*
+		 * Keep statistics on lines changed (if any).
+		 */
+		if ( strlen( $git_result_item ) > 0 ) {
```

Comment:
```suggestion
		if (
			strlen( $git_result_item ) > 0
			&& isset( VIPGOCI_GIT_DIFF_CALC_CHANGES[ $git_result_item[0] ] )
		) {
```
We could also move the ``isset( VIPGOCI_GIT_DIFF_CALC_CHANGES[ $git_result_item[0] ] )`` condition to the first if condition with a ``&&``

Reply:
Resolved in 14e020e2